api/placer.py

```diff
@@ -674,3 +675,25 @@ def finalize(self):
 
             config.db.sessions.update_one(q, u)
             return self.saved
+
+class GearPlacer(Placer):
+    def check(self):
+        pass
+        # self.requireTarget()
+        # validators.validate_data(self.metadata, 'file.json', 'input', 'POST', optional=True)
+
+    def process_file_field(self, field, file_attrs):
+        if self.metadata:
+            file_attrs.update(self.metadata)
+            self.metadata.update({'exchange': {'rootfs-hash': file_attrs.get('hash'),
+                                               'git-commit': 'local',
+                                               'rootfs-url': '/api/gears/temp/'+file_attrs.get('hash')}})
+        # self.metadata['hash'] = file_attrs.get('hash')
```

Comment:
Since you're doing everything you need to with the `file_attrs` here, as I mentioned in the earlier comment, no need to pass them to `save_file`.